src/concrete/ml/deployment/fhe_client_server.py

```diff
@@ -238,34 +266,17 @@ def load(self):  # pylint: disable=no-value-for-parameter
         Raises:
             ValueError: if mismatch in versions between serialized file and runtime
         """
-        self.client = fhe.Client.load(Path(self.path_dir).joinpath("client.zip"), self.key_dir)
+        client_zip_path = Path(self.path_dir).joinpath("client.zip")
+
+        self.client = fhe.Client.load(client_zip_path, self.key_dir)
 
         # Load the quantizers
-        with zipfile.ZipFile(Path(self.path_dir).joinpath("client.zip")) as client_zip:
+        with zipfile.ZipFile(client_zip_path) as client_zip:
             with client_zip.open("serialized_processing.json", mode="r") as file:
                 serialized_processing = load(file)
 
-        # Load versions for checking
-        with zipfile.ZipFile(Path(self.path_dir).joinpath("client.zip")) as client_zip:
-            with client_zip.open("versions.json", mode="r") as file:
-                versions = json.load(file)
-
-        errors = []
-        packages_to_check = {"concrete-python", "concrete-ml"}
-        for package_name, package_version in versions.items():
-            if package_name not in packages_to_check:
-                continue
-            if package_name == "concrete-ml":
-                current_version = CML_VERSION
-            else:
-                current_version = version(package_name)
-            if package_version != current_version:  # pragma: no cover
-                errors.append((package_name, package_version, current_version))
-        if errors:  # pragma: no cover
-            raise ValueError(
-                "Version mismatch for packages: \n"
-                + "\n".join(f"{error[0]}: {error[1]} != {error[2]}" for error in errors)
-            )
+        # Load and check versions
+        check_concrete_versions(client_zip_path)
 
         # Make sure the version in serialized_model is the same as CML_VERSION
         assert_true(
```

Comment:
is the following code still necessary ? it seems like Concrete ml's version is also tracked by the `serialized_processing` object, but it looks like it's redundant : I imagine this is kept  for backward compatibility ? @fd0r

Reply:
Hum I believe we should be able to remove this now.
Could be done in the PR if okay with you

Reply:
yes done !